src/sentry/db/postgres/transactions.py

```diff
@@ -22,21 +27,20 @@ def django_test_transaction_water_mark(using: str = "default"):
 
     from sentry.testutils import hybrid_cloud
 
-    # No need to manage the watermark unless conftest has configured a watermark
-    if using not in hybrid_cloud.simulated_transaction_watermarks.state:
-        yield
-        return
+    using = determine_using_by_silo_mode(using)
 
     connection = transaction.get_connection(using)
 
-    prev = hybrid_cloud.simulated_transaction_watermarks.state[using]
-    hybrid_cloud.simulated_transaction_watermarks.state[using] = len(connection.savepoint_ids)
+    prev = hybrid_cloud.simulated_transaction_watermarks.state.get(using, -1)
```

Comment:
We need transaction watermarks to flag calls to RPC services that occur inside of a transaction, to prevent a common performance issue that occurs when invoking long running IO inside of a transaction.  This logic used to only work for DjangoTestCase but not DjangoTransactionTestCase, but it has been extended to support both modes.

The trick here, the reason why the logic is annoying, is that tracking the number of outer transactions is painful.  For one, django connection has 2 pieces of state, `in_atomic_block` and `savepoint_ids`.  The integer of watermarks is for comparison against the later, but only when it is not -1 (indicating a check against the former).

Reply:
Should we have a name for `-1` to add semantics for future us?

Reply:
Yeah, I am also annoyed at how django manages transaction state.  Actually, your comment made me realize I could model this as a tuple `(connection.in_atomic_block, len(connection.savepoint_ids))` because incidentally, sorting tuples would have this work the way I need it to.